core/src/main/java/io/grpc/internal/ReadableBuffer.java

```diff
@@ -123,6 +124,39 @@
    */
   int arrayOffset();
 
+  /**
+   * Marks the current position in this buffer. A subsequent call to the {@link #reset} method
+   * repositions this stream at the last marked position so that subsequent reads re-read the same
+   * bytes.
+   */
+  void mark();
+
+  /**
+   * Repositions this buffer to the position at the time {@link #mark} was last called on this
+   * buffer.
+   */
+  void reset();
+
+  /**
+   * Indicates whether or not {@link #getByteBuffer} operation is supported for this buffer.
+   */
+  boolean canUseByteBuffer();
```

Comment:
nit: maybe better name would be `hasByteBuffer()`?

Reply:
Sure, go as you like. I don't have a strong preference.